### ISSUE TITLE
docs: provide guidance on how patterns, practices, things to avoid

### ISSUE THREAD

**Issue by EisenbergEffect** [microsoft/fast#4310]
We should take more of our learnings from building components and experience and produce documentation that leads customers into the pit of success.

* Project organization
* File structure
* Naming components/classes
* How to think about state
* TypeScript practices

**Comment by adamcee** [comment]
Agreed : ). Looking at FAST and finding it fascinating. Half of my curiosity is that it is, in the code, well-documented UI component behavior/state, independent of implementation.

**Comment by send2vinnie** [comment]
Pure Javascript practices such as how to work without decorators in the various topics are also suggested.
[DI without decorators](https://github.com/microsoft/fast/issues/6376)

**Comment by janechu** [comment]
This is being addressed in the documentation update to `@microsoft/fast-element` v2, including a component library section for developers looking to publish their own component packages.
